pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -171,46 +157,113 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 							return
 						case <-time.After(controller.NamespaceResyncPeriod):
 							newRootCert := s.CA.GetCAKeyCertBundle().GetRootCertPem()
-							if !bytes.Equal(rootCert, newRootCert) {
-								rootCert = newRootCert
-								if err = ioutil.WriteFile(internalSelfSignedRootPath, rootCert, 0o600); err != nil {
-									log.Errorf("Failed to update local copy of self-signed root: %v", err)
-								} else {
-									log.Info("Updated local copy of self-signed root")
-								}
+							if !bytes.Equal(caBundle, newRootCert) {
+								caBundle = newRootCert
```

Comment:
I am not sure about this. What does updating `caBundle` do? Its just a local variable - how will downstream components get notified?

Reply:
Done, added to keycert bundle watcher